python/cudf/cudf/_lib/utils.pyx

```diff
@@ -98,7 +99,14 @@ cpdef generate_pandas_metadata(Table table, index):
         ):
             types.append(col.dtype.to_arrow())
         else:
-            types.append(np_to_pa_dtype(col.dtype))
+            # A boolean element takes 8 bits in cudf and 1 bit in
+            # pyarrow. To make sure the cudf format is interperable
+            # in arrow, we use `int8` type when converting from a
+            # cudf boolean array.
+            if col.dtype.type == np.bool_:
+                types.append(pa.int8())
+            else:
+                types.append(np_to_pa_dtype(col.dtype))
```

Comment:
Instead of this we could consider having a private dict that handles this difference in key. Looks like this logic is repeated below as well. This was we could also change the dictionary in the future if we ever changed our boolean representation to use a bit per value (please no...) and wouldn't have to track all of these throughout the codebase.

Reply:
The difficult thing is to encapsulate the idea of "every dtype should map, except the one that has a different bitwidth` into a descriptive name. We were recently just discussing about indirection and [this](https://matthewrocklin.com/blog/work/2019/06/23/avoid-indirection) suggest bad namings may raise difficulty in reviews.

Reply:
And... hopefully we shouldn't need to have this logic all over the code base as this is only used in device->host logic. While `np_to_pa_dtype` maybe used everywhere.

cc. @vyasr on recent refactorings.

Reply:
I don't feel super strongly about this, I just saw the logic repeated once already and could imagine it repeating elsewhere. We can always tackle that problem later.